### PR TITLE
[#1814] Initial combining of Zoink and Popup plugins, sans transitions/fonts to be handled later

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1814-create-new-popup-plugin

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/a73aa863d0200bc/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/5ac5e9cc8bccd50/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/a73aa863d0200bc/output.txt

Total script time: 0.54 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5ac5e9cc8bccd50/output.txt

Total script time: 0.55 mins

Server is running on http://pr701.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/eb2ffa9cc298f36/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/eb2ffa9cc298f36/output.txt

Total script time: 1.02 mins

Server is running on http://pr701.butyr.org/templates/test.html

File: templates/assets/plugins/popup/popcorn.popup.js [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+      }
+
+      options._target = target;
+
+      if ( options.classes === "none" ) {
+        options.classes = "";
+      }
+
+      var style = container.style,
+          flip = options.flip && " flip" || "",
```
Comment (secretrobotron) [inline templates/assets/plugins/popup/popcorn.popup.js]: Can you turn this into an actual `?` conditional statement? This looks a bit hacky.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f6af1314628c4de/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f6af1314628c4de/output.txt

Total script time: 0.98 mins

Server is running on http://pr701.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/09ef535cb70f1e1/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/09ef535cb70f1e1/output.txt

Total script time: 0.81 mins

Server is running on http://pr701.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d51ada4fead2866/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d51ada4fead2866/output.txt

Total script time: 0.81 mins

Server is running on http://pr701.butyr.org/templates/basic/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/12764949fbe4092/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/12764949fbe4092/output.txt

Total script time: 0.37 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/dd7988fae442f09/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/dd7988fae442f09/output.txt

Total script time: 0.69 mins

Server is running on http://pr701.butyr.org/templates/basic/
Run the Test Suite: http://pr701.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/fb9e608cf32d02f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/fb9e608cf32d02f/output.txt

Total script time: 1.07 mins

Server is running on http://pr701.butyr.org/templates/basic/
Run the Test Suite: http://pr701.butyr.org/test/

File: templates/assets/editors/popup.js [hunk 2]
```diff
@@ -12,5 +12,6 @@ EditorHelper.addPlugin( "zoink", function( trackEvent ) {
 
   if ( window.jQuery ) {
     window.EditorHelper.draggable( trackEvent, _container, media );
+    window.EditorHelper.resizable( trackEvent, _container, media, "e" );
```
Comment (k88hudson) [inline templates/assets/editors/popup.js]: Remove `window`

File: templates/assets/plugins/popup/popcorn.popup.js [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+
+  drawSpeech( innerDivTriangles.speech, 2 );
+  drawThought( innerDivTriangles.thought, 2 );
+
+  Popcorn.plugin( "popup", {
+    manifest: {
+      about: {
+        name: "Popcorn Maker Popup Plugin",
+        version: "0.1",
+        author: "Kate Hudson @k88hudson, Matthew Schranz @mjschranz",
```
Comment (k88hudson) [inline templates/assets/plugins/popup/popcorn.popup.js]: Add brian chirls here

**Pull request merged**